src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.OrderItemResponseDto;
+import com.example.onlinebookstore.dto.OrderRequestDto;
+import com.example.onlinebookstore.dto.OrderResponseDto;
+import com.example.onlinebookstore.dto.OrderStatusUpdateDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.Status;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.service.OrderService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public OrderResponseDto placeOrder(User user, OrderRequestDto orderRequestDto) {
```

Comment:
we should also clear the list of cartItems for the shopping cart of that user when we have placed an order.
Also it is vital to have it as an atomic operation, because we do not want to clear cart items if we have encountered some error with creating an order, so:
```suggestion
    @Override
    @Transactional
    public OrderResponseDto placeOrder(User user, OrderRequestDto orderRequestDto) {
```
And take a look where you might want to add it for other methods as well, maybe with `readOnly = true` param, to boost the performance of get operations?

Reply:
done